TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/RoadSystemGraph.cs

```diff
@@ -40,32 +40,41 @@ public class RoadNavigationGraph
         };
 
         private float _currentCost = 0f;
-        public RoadNavigationGraph(RoadNode roadNode)
+        public RoadNavigationGraph(RoadNode roadNode, bool isClosed)
         {
             RoadNode curr = roadNode;
             GraphNode PreviouslyAddedNode = null;
+            // If it is an closed road we don't want to add the start node to the graph, so we skip it
+            if (isClosed)
+                curr = curr.Next;
             while (curr != null)
             {
-                // The start node is added to the graph
-                if (PreviouslyAddedNode == null)
+                // Increase the current cost if the current node is not the starting node
+                if (curr.Prev != null)
+                    _currentCost += CalculateCost(Vector3.Distance(curr.Position, curr.Prev.Position));
```

Comment:
Should use `curr.DistanceToPrevNode` from https://github.com/marc7s/TrafficSimulator/pull/35 once that is merged